### PR TITLE
chore: Consistent/Improved logging

### LINKED ISSUE #124: Update logging of external-dns code
**What**

We currently have code copied and modified from external-dns which uses the logrus logger. We don't really want to change this since we hope to eventually remove it , but we do need all the logs it produces to go through the same logger as everything else in the dns operator.

**How**

Use a hook and catch all logrus logs and pass them to the dns operator logs?

### PR DESCRIPTION
**Pull request by mikenairn**
closes #124 

[external-dns: Remove logrus logger](https://github.com/Kuadrant/dns-operator/commit/55bf01a274f7e927afba9d03a9fbada88d8f8050)

Removes the use of logrus in all external-dns code in favour of setting a `logr.Logger` on all objects.

Constructors are used throughout to create new instances of components (plan, registry and providers), with each requiring the current context,  retrieving a logger instance from it with an appropriate logger name added where required.

[chore: Consistent/Improved logging](https://github.com/Kuadrant/dns-operator/commit/c4d27a924e6885a9d376239ec1bcdd759d781713)

Construct logger in `Reconcile` functions (DNSRecord and ManagedZone), naming each appropriately, and ensure it is used by all logging statements throughout the call stack for the request.

Initialise external-dns components (registry, plan and providers) using correct context to ensure the appropriate logger is used internally. Removes all references to logrus logger.

### CODE REVIEW

File: internal/controller/managedzone_controller.go [hunk 1]
```diff
@@ -156,7 +157,7 @@ func (r *ManagedZoneReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
-	log.Log.Info("Reconciled ManagedZone", "managedZone", managedZone.Name)
+	logger.Info("Reconciled ManagedZone")
```
Comment (philbrookes) [inline internal/controller/managedzone_controller.go:160]: why don't we need the managedzone name here, but we do need it in some of the others?

Comment (philbrookes) [reply internal/controller/managedzone_controller.go:160]: like here: https://github.com/Kuadrant/dns-operator/pull/152/files#diff-5882be9025c1bbbef34525f6ffba174154eaac4c288cd24fb8ebb968a287bf84R220

Comment (mikenairn) [reply internal/controller/managedzone_controller.go:160]: We don't will update those to remove the duplicate key/values

File: internal/controller/dnsrecord_controller.go [hunk 2]
```diff
@@ -271,15 +282,15 @@ func (r *DNSRecordReconciler) deleteRecord(ctx context.Context, dnsRecord *v1alp
...
 			return false, nil
 		} else if strings.Contains(err.Error(), "no endpoints") {
-			logger.Info("DNS record had no endpoint, continuing", "dnsRecord", dnsRecord.Name, "managedZone", managedZone.Name)
+			logger.Info("DNS record had no endpoint, continuing", "managedZone", managedZone.Name)
 			return false, nil
 		}
 		return false, err
 	}
-	logger.Info("Deleted DNSRecord in manage zone", "dnsRecord", dnsRecord.Name, "managedZone", managedZone.Name)
+	logger.Info("Deleted DNSRecord in manage zone", "managedZone", managedZone.Name)
```
Comment (mikenairn) [inline internal/controller/dnsrecord_controller.go:293]: Example:

```.json
{
  "level": "info",
  "ts": "2024-06-04T16:32:37+01:00",
  "logger": "dnsrecord_controller",
  "msg": "Deleted DNSRecord in manage zone",
  "controller": "dnsrecord",
  "controllerGroup": "kuadrant.io",
  "controllerKind": "DNSRecord",
  "DNSRecord": {
    "name": "t-single-lively-waterfall",
    "namespace": "dnstest"
  },
  "namespace": "dnstest",
  "name": "t-single-lively-waterfall",
  "reconcileID": "81a1838b-c782-4323-8baf-58486cadbcec",
  "managedZone": "dev-mz-aws"
}
```

Left `managedZone` here for now although i expect that will be moved and set on the context higher up the stack as part of https://github.com/Kuadrant/dns-operator/issues/117

File: internal/controller/managedzone_controller.go (hunk 1 above)
Comment (mikenairn) [reply internal/controller/managedzone_controller.go:160]: Should all be updated now to remove adding the details of the current resource at individual log statements. That is already being done as part of the logger setup in `Reconcile`. Further improvements to the logging and what metadata is included and where will be done as part of  https://github.com/Kuadrant/dns-operator/issues/118

**Pull request merged**